### DISCUSSION TITLE
Is it possible  to inject ---- android:text="@string/login"

### DISCUSSION THREAD

**Discussion by AndroXStudiO** [Sketchware-Pro/Sketchware-Pro#655] [Q&A] [answered]
Is it possible  to inject ---- android:text="@string/login"  or 
 android:textSize="@dimens/12dp"------ in TextView?
because i want to make multiple language app...and so, i have to use those attitude.

**Comment by j-px** [comment, marked as answer]
These ways exist: (use v6.4.0-rc05 at best for the latest features)
 - XML Command Blocks: work even in v6.3.0 fix1 and earlier. [More details on them here in our Discord server.](https://discord.com/channels/790686719753846785/797801500276424724/798164544375619625)
 - Entering e.g. just `@string/login` as text in the TextView/EditText/etc.: needs v6.4.0
 - Adding e.g. `tools:replace="android:textSize" android:textSize="@dimen/text_size_login"` to the inject property. That'll replace `android:textSize` added with the text size property and make it use the injected one instead. Needs v6.4.0-rc05 or later.

**Comment by AndroXStudiO** [comment]
Help me if possible..!!

**Reply by j-px** [reply]
I just did. Those are the ways you can do that.

**Comment by AndroXStudiO** [comment]
![Screenshot_2022-11-24-22-17-41-25_423e58f25c30a02453fa5d8c82fc37f2](https://user-images.githubusercontent.com/88816748/203829295-eddd1e70-d0bc-4e30-9631-7b40427b1f2e.jpg)


not Working

**Comment by AndroXStudiO** [comment]
![Screenshot_2022-11-24-22-17-48-34_423e58f25c30a02453fa5d8c82fc37f2](https://user-images.githubusercontent.com/88816748/203829362-dddc4ee6-9b3d-4d6c-bc31-a881d2c156b0.jpg)

**Comment by AndroXStudiO** [comment]
![Screenshot_2022-11-24-22-19-26-96_423e58f25c30a02453fa5d8c82fc37f2](https://user-images.githubusercontent.com/88816748/203829591-1e3ef4a8-bc53-4ea8-adb1-398c533e2f5c.jpg)

**Reply by j-px** [reply]
Upload main.xml as file here.

**Reply by AndroXStudiO** [reply]
<androidx.coordinatorlayout.widget.CoordinatorLayout
	xmlns:android="http://schemas.android.com/apk/res/android"
	xmlns:app="http://schemas.android.com/apk/res-auto"
	xmlns:tools="http://schemas.android.com/tools"
	android:id="@+id/_coordinator"
	android:layout_width="match_parent"
	android:layout_height="match_parent">
	<com.google.android.material.appbar.AppBarLayout
		android:id="@+id/_app_bar"
		android:layout_width="match_parent"
		android:layout_height="wrap_content"
		android:theme="@style/AppTheme.AppBarOverlay">
		<androidx.appcompat.widget.Toolbar
			android:id="@+id/_toolbar"
			android:layout_width="match_parent"
			android:layout_height="?attr/actionBarSize"
			android:background="?attr/colorPrimary"
			app:popupTheme="@style/AppTheme.PopupOverlay" />
	</com.google.android.material.appbar.AppBarLayout>
	<LinearLayout
		android:layout_width="match_parent"
		android:layout_height="match_parent"
		android:orientation="vertical"
		app:layout_behavior="@string/appbar_scrolling_view_behavior">
		<LinearLayout
			android:id="@+id/linear1"
			android:layout_width="match_parent"
			android:layout_height="match_parent"
			android:paddingLeft="8dp"
			android:paddingTop="8dp"
			android:paddingRight="8dp"
			android:orientation="vertical">
			<Button
				android:id="@+id/button1"
				android:layout_width="wrap_content"
				android:layout_height="wrap_content"
				android:padding="8dp"
				android:text="Button"
				android:textSize="12sp"
				android:textColor="#000000" />
			<EditText
				android:id="@+id/edittext1"
				android:layout_width="wrap_content"
				android:layout_height="wrap_content"
				android:padding="8dp"
				android:textSize="12sp"
				android:textColor="#000000"
				android:hint="Edit Text"
				android:textColorHint="#607D8B"
				android:inputType="phone"
				android:password="true" />
			<TextView
				android:id="@+id/textview1"
				android:layout_width="wrap_content"
				android:layout_height="wrap_content"
				android:padding="8dp"
				android:text="TextView"
				android:textColor="#000000"
				tools:replace="android:textSize"android:textSize="@dimen/text_size_login" />
		</LinearLayout>
	</LinearLayout>
</androidx.coordinatorlayout.widget.CoordinatorLayout>

**Reply by j-px** [reply]
Put a space after the 2nd `"` in the injected value.

**Reply by j-px** [reply]
Wait, not a space, but a newline/enter.

**Reply by AndroXStudiO** [reply]
its not possible directly?

**Comment by AndroXStudiO** [comment]
![Screenshot_2022-11-24-22-46-25-48_423e58f25c30a02453fa5d8c82fc37f2](https://user-images.githubusercontent.com/88816748/203834167-605d4fb4-63fa-4342-a8d1-f5cf35f3b23f.jpg)


See for textColor old version of sketchware pro...work directly..and is amazing

**Reply by j-px** [reply]
What exact old version?

**Comment by AndroXStudiO** [comment]
![Screenshot_2022-11-24-22-46-34-33_423e58f25c30a02453fa5d8c82fc37f2](https://user-images.githubusercontent.com/88816748/203834366-be16d5c7-efaf-422b-b660-b91c207ec1d7.jpg)


directly injected

**Comment by AndroXStudiO** [comment]
@JavkhlanK  
![Screenshot_2022-11-24-22-51-39-25_423e58f25c30a02453fa5d8c82fc37f2](https://user-images.githubusercontent.com/88816748/203834967-01392a03-c1f6-4a99-afc6-46866a007d1b.jpg)


i think something went wrong...see this picture. 
android:textColor --- i injected by own.and its replace automatically. 
but when i added android:textSize --- its not replace, it got duplicate..... why??

**Reply by j-px** [reply]
Some get automatically not added if it's detected that the inject property contains them.

Anyway, you must add `tools:replace="android:textColor"` also to inject, then press enter, and then the rest of the inject property!

**Reply by AndroXStudiO** [reply]
Not Working for me

**Reply by j-px** [reply]
Did you press enter / have a newline between `tools="…"` and `android:textSize="…"` in the inject property?

**Reply by AndroXStudiO** [reply]
Yes...now working ❤️. But it would be great if it was automatic...
